source/iofs/iofs.go

```diff
@@ -0,0 +1,166 @@
+// +build go1.16
+
+package iofs
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"io/fs"
+	"path"
+	"strconv"
+
+	"github.com/golang-migrate/migrate/v4/source"
+)
+
+// Driver is a source driver that wraps io/fs#FS.
+type Driver struct {
+	migrations *source.Migrations
+	fsys       fs.FS
+	path       string
+}
+
+// NewDriver returns a new Driver from io/fs#FS and a relative path.
+func NewDriver(fsys fs.FS, path string) (source.Driver, error) {
+	var i Driver
+	if err := i.Init(fsys, path); err != nil {
+		return nil, fmt.Errorf("failed to init driver with path %s: %w", path, err)
+	}
+	return &i, nil
+}
+
+// Open is part of source.Driver interface implementation.
+// Open panics when called directly.
+func (d *Driver) Open(url string) (source.Driver, error) {
+	panic("iofs: Driver does not support open with url")
+}
+
+// Init prepares not initialized IoFS instance to read migrations from a
+// io/fs#FS instance and a relative path.
+func (d *Driver) Init(fsys fs.FS, path string) error {
```

Comment:
Move all of the driver receiver methods (except for `Open()`) to `PartialDriver` to follow the same patter than [httpfs](https://pkg.go.dev/github.com/golang-migrate/migrate/v4@v4.13.0/source/httpfs#PartialDriver) uses.